reporters/kamon-opentelemetry/src/main/scala/kamon/otel/MetricsConverter.scala

```diff
@@ -0,0 +1,121 @@
+package kamon.otel
+
+import io.opentelemetry.sdk.common.InstrumentationScopeInfo
+import io.opentelemetry.sdk.metrics.data._
+import io.opentelemetry.sdk.metrics.internal.data._
+import io.opentelemetry.sdk.resources.Resource
+import kamon.metric.Instrument.Snapshot
+import kamon.metric.{Distribution, MeasurementUnit, MetricSnapshot, PeriodSnapshot}
+import kamon.tag.Lookups
+import kamon.trace.Span.TagKeys
+
+import java.lang.{Double => JDouble, Long => JLong}
+import java.time.Instant
+import java.util.{Collection => JCollection}
+import scala.collection.JavaConverters._
+import scala.collection.mutable.ArrayBuffer
+
+class WithResourceMetricsConverter(resource: Resource, kamonVersion: String, from: Instant, to: Instant) {
+  private val fromNs = from.toEpochMilli * 1000000
+  private val toNs = to.toEpochMilli * 1000000
+
+  private def instrumentationScopeInfo(snapshot: MetricSnapshot[_, _]): InstrumentationScopeInfo = {
+    // logic for looking up the component doesn't really seem to make sense - to be compliant we should probably be grouping the metrics by component before calling this
+    InstrumentationScopeInfo.create(snapshot.instruments.headOption.flatMap(_.tags.get(Lookups.option(TagKeys.Component))) getOrElse "kamon-instrumentation", kamonVersion, null)
+  }
+
+  private def toString(unit: MeasurementUnit): String = unit.magnitude.name
+
+  def toGaugeDatum(g: Snapshot[Double]): DoublePointData = ImmutableDoublePointData.create(fromNs, toNs, SpanConverter.toAttributes(g.tags), g.value)
+
+  def toGaugeData(g: Seq[Snapshot[Double]]): GaugeData[DoublePointData] = ImmutableGaugeData.create(g.map(toGaugeDatum).asJava)
+
+  def convertGauge(gauge: MetricSnapshot.Values[Double]): MetricData =
+    ImmutableMetricData.createDoubleGauge(
+      resource,
+      instrumentationScopeInfo(gauge),
+      gauge.name,
+      gauge.description,
+      toString(gauge.settings.unit),
+      toGaugeData(gauge.instruments))
+
+  def toHistogramDatum(s: Snapshot[Distribution]): HistogramPointData = {
+    val boundaries = ArrayBuffer.newBuilder[JDouble]
+    val counts = ArrayBuffer.newBuilder[JLong]
+    for (el <- s.value.bucketsIterator) {
+      counts += el.frequency
+      boundaries += el.value.toDouble
+    }
+    ImmutableHistogramPointData.create(
+      fromNs,
+      toNs,
+      SpanConverter.toAttributes(s.tags),
+      JDouble valueOf s.value.sum.toDouble,
+      JDouble valueOf s.value.min.toDouble,
+      JDouble valueOf s.value.max.toDouble,
+      boundaries.result().dropRight(1).asJava,
+      counts.result().asJava
+    )
+  }
+
+  def toHistogramData(distributions: Seq[Snapshot[Distribution]]): Option[HistogramData] =
+    distributions.filter(_.value.buckets.nonEmpty) match {
+      case Nil => None
+      case nonEmpty => Some(ImmutableHistogramData.create(AggregationTemporality.CUMULATIVE, nonEmpty.map(toHistogramDatum).asJava))
+    }
+
+  def convertHistogram(histogram: MetricSnapshot.Distributions): Option[MetricData] =
```

Comment:
Have you considered using exponential histograms instead of the plain base histogram here? Exponential histograms would provide a much closer representation of Kamon's HDR histogram data

Reply:
I had not... Hadn't even looked at it, in fact. Will take a look later on today

Reply:
Refactored this pr to support `OTEL_EXPORTER_OTLP_METRICS_DEFAULT_HISTOGRAM_AGGREGATION` (in theory)